In [1]:
from cvfwVer2 import CVFW_MODEL, CVFW_UPDATE

cvfw_model = CVFW_MODEL(dsize=(100, 100))

cvfw_model.add_directory(class_name="female face", path="C:\\kimdonghwan\\python\\CVFW\\image\\train\\face\\femaleface")
cvfw_model.add_directory(class_name="male face", path="C:\\kimdonghwan\\python\\CVFW\\image\\train\\face\\maleface")

In [2]:
cvfw_model.train()

male face: 100%|███████████████████████████████████████████████████████████████| 10000/10000 [00:07<00:00, 1273.31it/s]


In [3]:
from os import listdir

print(len(listdir("C:\\kimdonghwan\\python\\CVFW\\image\\train\\face\\femaleface")))
print(len(listdir("C:\\kimdonghwan\\python\\CVFW\\image\\train\\face\\maleface")))

461
390


In [6]:
cvfw_update = CVFW_UPDATE(cvfw_model, feature_group_number=[100, 500], feature_weight_number=[30, 50])
cvfw_update.add_validation(class_name="female face", path = "./image/test/face/femaleface")
cvfw_update.add_validation(class_name="male face", path = "./image/test/face/maleface")
cvfw_update.update()

feature_group_number: 100, feature_weight_nummber: 30 Done!, accuracy: 0.6454183266932271
feature_group_number: 100, feature_weight_nummber: 50 Done!, accuracy: 0.6653386454183267
feature_group_number: 500, feature_weight_nummber: 30 Done!, accuracy: 0.6454183266932271
feature_group_number: 500, feature_weight_nummber: 50 Done!, accuracy: 0.6653386454183267


In [7]:
cvfw_update.set(feature_group_number = [20, 50, 70], feature_weight_number = [30, 50, 75])
cvfw_update.update()

feature_group_number: 20, feature_weight_nummber: 30 Done!, accuracy: 0.6454183266932271
feature_group_number: 20, feature_weight_nummber: 50 Done!, accuracy: 0.6653386454183267
feature_group_number: 20, feature_weight_nummber: 75 Done!, accuracy: 0.701195219123506
feature_group_number: 50, feature_weight_nummber: 30 Done!, accuracy: 0.6454183266932271
feature_group_number: 50, feature_weight_nummber: 50 Done!, accuracy: 0.6653386454183267
feature_group_number: 50, feature_weight_nummber: 75 Done!, accuracy: 0.701195219123506
feature_group_number: 70, feature_weight_nummber: 30 Done!, accuracy: 0.6454183266932271
feature_group_number: 70, feature_weight_nummber: 50 Done!, accuracy: 0.6653386454183267
feature_group_number: 70, feature_weight_nummber: 75 Done!, accuracy: 0.701195219123506


### 최적의 feature_group_number, feature_weight_number
- feature_group_number: 20
- feature_weight_number: 75
- accuracy: 70.1195219123506%